figure_generation/nbconverted/figure_generation.py

```diff
@@ -20,8 +20,7 @@
 
 # Directory of output figures
 output_directory = "output/"
-if not os.path.exists(output_directory):
-    os.makedirs(output_directory)
+os.makedirs(output_directory, exist_ok=True)
```

Comment:
It looks like some of the changes in this notebook were made in your last PR (or at least I remember reviewing them before). Maybe merging in your `master` branch would help?

Reply:
So these changes are because I merged in my master branch with this one to resolve some conflicts. So those are indeed changes from previously merged in PRs.